src/jaxsim/api/ode.py

```diff
@@ -150,14 +150,12 @@ def system_velocity_dynamics(
         # Sum the forces of all collidable points rigidly attached to a body.
         # Since the contact forces W_f_Ci are expressed in the world frame,
         # we don't need any coordinate transformation.
-        W_f_Li_terrain = jax.vmap(
-            lambda nc: (
-                jnp.vstack(
-                    jnp.equal(parent_link_index_of_collidable_points, nc).astype(int)
-                )
-                * W_f_Ci
-            ).sum(axis=0)
-        )(jnp.arange(model.number_of_links()))
+        W_f_Li_terrain = jnp.where(
+            parent_link_index_of_collidable_points[:, jnp.newaxis]
+            == jnp.arange(model.number_of_links()),
+            W_f_Ci,
+            0.0,
```

Comment:
Shouldn't this be a `jnp.zeros(6)`?

Reply:
Yes, thanks a lot!

Reply:
Edited my suggestion, you replied too quickly :)

Reply:
Done in https://github.com/ami-iit/jaxsim/pull/180/commits/9f7cdf546efb3f087cd4867e177cc4330768e875

Reply:
Before merging, please run a free-falling simulation with a box. Probably we can add in the future this simple test in `test_simulations.py` since we don't have anything that checks a contact-aware simulation.

Reply:
Using one of the example notebook: 

![image](https://github.com/ami-iit/jaxsim/assets/102977828/61578f1c-bd13-411e-948e-c2e8748f5465)